ReturnNonDuplicateArrays.js

```diff
@@ -0,0 +1,15 @@
+function returnCapabilities(jobCapability){
+     //collect all capabilities into set to remove duplicates
+    return new Set(jobCapability.map(function(object){
+        return object.capability;
+    }))
+}
+
+function returnBandLevels(bandLevels){
+    return new Set(bandLevels.map(function(object){
+        return object.bandLevel;
+    }))
+}
+
+module.exports.returnCapabilities = returnCapabilities
+module.exports.returnBandLevels = returnBandLevels
```

Comment:
This approach means you'll need to add a new method every time you want to remove duplicates from an array of a new type, which could get unwieldy. You could instead provide a single method which lets the caller define how the  input array should be mapped

```suggestion
function mapDistinct(inputs, mapper) {
    return new Set(inputs.map(mapper))
}
module.exports = { mapDistinct }
```

Then it would be called like `mapDistinct(jobs, job => job.capability)`

What do you think?

Reply:
I think it's a much cleaner approach, even though I think we'll only need these two it's always better for it to be extensible so I'll have a look at changing it. Thanks for the suggestion